core/docker/default/etc/jvm.config

```diff
@@ -15,3 +15,5 @@
 # Reduce starvation of threads by GClocker, recommend to set about the number of cpu cores (JDK-8192647)
 -XX:+UnlockDiagnosticVMOptions
 -XX:GCLockerRetryAllocationCount=32
+# Allow loading dynamic agent used by JOL
+-XX:+EnableDynamicAgentLoading
```

Comment:
According to doc proper approach would be to add manifest entries in final jar:
```
<manifestEntries>
                                        <Premain-Class>org.openjdk.jol.vm.InstrumentationSupport</Premain-Class>
                                        <Launcher-Agent-Class>org.openjdk.jol.vm.InstrumentationSupport$Installer</Launcher-Agent-Class>
                                    </manifestEntries>
```

please create an issue and and todo comment above

Reply:
I've tried that, with no luck. I've filled https://github.com/trinodb/trino/issues/20904